octobot_trading/modes/modes_util.py

```diff
@@ -45,3 +53,108 @@ async def clear_plotting_cache(trading_mode):
+        -> (list, dict):
+    created_orders = []
+    for asset in sellable_assets:
+        new_orders, tickers = await convert_asset_to_target_asset(
+            trading_mode, asset, target_asset, asset_amount=None, tickers=tickers
+        )
+        created_orders += new_orders
+    return created_orders, tickers
+
+
+async def convert_asset_to_target_asset(
+    trading_mode, asset: str, target_asset: str, asset_amount=None, tickers=None
+) -> (list, dict):
+    if asset == target_asset:
+        return [], tickers
+    created_orders = []
+    tickers = tickers or {}
+    portfolio = trading_mode.exchange_manager.exchange_personal_data.portfolio_manager.portfolio.portfolio
+    if asset in portfolio and portfolio[asset].available:
+        symbol = symbol_util.merge_currencies(asset, target_asset)
+        order_type = trading_enums.TraderOrderType.SELL_MARKET
+        if symbol not in trading_mode.exchange_manager.client_symbols:
+            # try reversed
+            reversed_symbol = symbol_util.merge_currencies(target_asset, asset)
+            if reversed_symbol not in trading_mode.exchange_manager.client_symbols:
+                # can't convert asset into target_asset
+                trading_mode.logger.error(
+                    f"Impossible to convert {asset} into {target_asset}: no {symbol} or "
+                    f"{reversed_symbol} trading pair on {trading_mode.exchange_manager.exchange_name}"
+                )
+                return created_orders, tickers
+            symbol = reversed_symbol
+            order_type = trading_enums.TraderOrderType.BUY_MARKET
+        # 1. try with converter
+        try:
+            price = trading_mode.exchange_manager.exchange_personal_data.portfolio_manager.\
+                portfolio_value_holder.value_converter.evaluate_value(
+                    asset, constants.ONE, raise_error=True, target_currency=target_asset, init_price_fetchers=False
+                )
+            if order_type is trading_enums.TraderOrderType.BUY_MARKET:
+                price = constants.ONE / price
+        except errors.MissingPriceDataError:
+            # 2. try with tickers
+            if not tickers:
+                # price not available, fetch it once with ticker
+                tickers = await trading_mode.exchange_manager.exchange.get_all_currencies_price_ticker()
+            try:
+                price = decimal.Decimal(str(
+                    tickers[symbol][trading_enums.ExchangeConstantsTickersColumns.CLOSE.value]
+                    or tickers[symbol][trading_enums.ExchangeConstantsTickersColumns.PREVIOUS_CLOSE.value]
+                ))
+            except KeyError:
+                # can't get price, should not happen as symbol is in client_symbols
+                trading_mode.logger.error(
+                    f"Impossible to convert {asset} into {target_asset}: {symbol} ticker can't be fetched"
+                )
+                return created_orders, tickers
```

Comment:
I don't understand, we are trying to get the price of a traded symbol here?

Reply:
if it's a traded symbol, price will be foundin "1. try with converter", however when rebalacing we might need price of symbols we are not trading (if we want to sell USD into USDT and we don't have any /USD trading pair)

Reply:
Yes ok, I was wondering why we have to fetch tickers. Does it exist scenarios where the ticker is not fetch or it will always be fetched ? I'm asking because I think if it's always used we should pass it as a param of the function instead of fetching them inside the function

Reply:
ticker is only fetched when trading assets we can't get the price of otherwise (it is passed as a param and only fetched once)